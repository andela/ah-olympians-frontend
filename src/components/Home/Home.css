.App-header {
  background-color: #F7F9F9;
  margin-top: 200px;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: black; }

.image-head {
  width: 100%;
  margin-top: 110px; }

.carousel {
  height: 400px;
  margin: 0 auto; }

.carousel-item {
  height: 400px; }

.carousel-item img {
  object-fit: cover; }

.card-text {
  font-size: small; }
