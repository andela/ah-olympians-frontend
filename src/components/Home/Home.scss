@import "../../css/variables";

.App-header {
  background-color: #282c34;
  margin-top: 200px;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: $white-color; }
