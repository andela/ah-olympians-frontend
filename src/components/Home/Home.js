import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  Button,
  Card,
  CardColumns,
  Carousel,
  Container,
  Image,
  Row,
} from 'react-bootstrap';
import NavbarInstance from '../Navbar/Navbar';
import Footer from '../static/Footer';
import coffee from './img/coffee.jpg';
import create from './img/create.jpg';
import notebook from './img/notebook.jpg';
import { getArticles } from '../../actions/getArticles';
import './Home.scss';
import '../../App.scss';
import '../../css/bootstrap.min.css';

export class HomePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      articles: [],
    };
  }

  componentWillMount() {
    this.props.getArticles();
  }

  componentWillReceiveProps(newProps) {
    this.setState({
      articles: newProps.articles.articles,
    });
  }

  render() {
    let articlesArray = this.state.articles;
    const arrayLength = articlesArray.length;

    if (arrayLength > 6) {
      articlesArray = articlesArray.slice(Math.max(arrayLength - 6, 1));
    }
    const articleItem = articlesArray.map(article => (
      <Card key={article.slug}>
        <Card.Img variant="top" src={coffee} />
        <Card.Body>
          <Card.Title>{article.title}</Card.Title>
          <Card.Text>Author -{article.author.username}</Card.Text>
          <Button
            variant="primary"
            onClick={function(slug) {
              this.props.history.push(`/article/${article.slug}`);
            }.bind(this)}
          >
            Read More...
          </Button>
        </Card.Body>
      </Card>
    ));
    return (
      <div className="App-header">
        <header>
          <NavbarInstance />
        </header>
        <div>
          <Container>
            <Row className="">
              <Carousel>
                <Carousel.Item>
                  <Image
                    className="d-block w-100 img-responsive"
                    src={coffee}
                    alt="First slide"
                    thumbnail
                  />
                  <Carousel.Caption>
                    <h3>First slide label</h3>
                    <p>
                      Nulla vitae elit libero, a pharetra augue mollis interdum.
                    </p>
                  </Carousel.Caption>
                </Carousel.Item>
                <Carousel.Item>
                  <Image
                    className="d-block w-100"
                    src={create}
                    alt="First slide"
                    thumbnail
                  />
                  <Carousel.Caption>
                    <h3>First slide label</h3>
                    <p>
                      Nulla vitae elit libero, a pharetra augue mollis interdum.
                    </p>
                  </Carousel.Caption>
                </Carousel.Item>
                <Carousel.Item>
                  <Image
                    className="d-block w-100"
                    src={notebook}
                    alt="First slide"
                    thumbnail
                  />
                  <Carousel.Caption>
                    <h3>First slide label</h3>
                    <p>
                      Nulla vitae elit libero, a pharetra augue mollis interdum.
                    </p>
                  </Carousel.Caption>
                </Carousel.Item>
              </Carousel>
            </Row>
          </Container>
          <br />
          <Container>
            <CardColumns>{articleItem}</CardColumns>
          </Container>
        </div>
        <br />
        <div className="footer-space" />
        <Footer />
      </div>
    );
  }
}

HomePage.propTypes = {
  getArticles: PropTypes.func.isRequired,
  articles: PropTypes.array,
};

const mapStateToProps = state => ({
  articles: state.articles.items,
});
export default connect(
  mapStateToProps,
  { getArticles },
)(HomePage);
