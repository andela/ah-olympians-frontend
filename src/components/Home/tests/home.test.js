import React from 'react';
import { shallow } from 'enzyme';
import { HomePage } from '../Home';


function setup() {
  const props = {
    getArticles: jest.fn(),
  };

  const enzymeWrapper = shallow(<HomePage {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}

describe('components', () => {
  describe('HomePage', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();

      expect(enzymeWrapper.find('header').hasClass('header')).toBe(false);

      expect(enzymeWrapper.find('CardColumns').text()).toBe('');

      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
