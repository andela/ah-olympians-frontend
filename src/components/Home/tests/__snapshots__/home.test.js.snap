// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components HomePage should render self and subcomponents 1`] = `
<div>
  <header>
    <ConnectFunction />
  </header>
  <div
    className="App-header"
  >
    <div>
      <Container
        as="div"
        fluid={false}
      >
        <Bootstrap(Row)
          className=""
        >
          <Bootstrap(undefined)>
            <CarouselItem>
              <Bootstrap(Image)
                alt="First slide"
                className="d-block w-100 img-responsive"
                src="coffee.jpg"
                thumbnail={true}
              />
              <CarouselCaption>
                <h3>
                  First slide label
                </h3>
                <p>
                  Nulla vitae elit libero, a pharetra augue mollis interdum.
                </p>
              </CarouselCaption>
            </CarouselItem>
            <CarouselItem>
              <Bootstrap(Image)
                alt="First slide"
                className="d-block w-100"
                src="create.jpg"
                thumbnail={true}
              />
              <CarouselCaption>
                <h3>
                  First slide label
                </h3>
                <p>
                  Nulla vitae elit libero, a pharetra augue mollis interdum.
                </p>
              </CarouselCaption>
            </CarouselItem>
            <CarouselItem>
              <Bootstrap(Image)
                alt="First slide"
                className="d-block w-100"
                src="notebook.jpg"
                thumbnail={true}
              />
              <CarouselCaption>
                <h3>
                  First slide label
                </h3>
                <p>
                  Nulla vitae elit libero, a pharetra augue mollis interdum.
                </p>
              </CarouselCaption>
            </CarouselItem>
          </Bootstrap(undefined)>
        </Bootstrap(Row)>
      </Container>
      <br />
      <Container
        as="div"
        fluid={false}
      >
        <CardColumns />
      </Container>
    </div>
  </div>
  <br />
  <div
    className="footer-space"
  />
  <Footer />
</div>
`;
